ProteoformSuiteInternal/Lollipop.cs

```diff
@@ -527,6 +532,16 @@ public static void get_theoretical_proteoforms(string current_directory)
             }
         }
 
+        //Generate lookup table for ptm sets based on rounded mass of eligible PTMs -- used in forming ET relations
+        public static void make_ptmset_dictionary()
+        {
+            foreach (PtmSet set in all_possible_ptmsets.Where(s => s.ptm_combination.Count == 1 || !s.ptm_combination.Select(ptm => ptm.modification).Any(m => m.monoisotopicMass == 0)))
+            {
+                if (all_possible_ptmset_dictionary.ContainsKey(Convert.ToInt32(set.mass))) all_possible_ptmset_dictionary[Convert.ToInt32(set.mass)].Add(set);
```

Comment:
Oh, interesting! Convert.ToInt32 [does round](https://msdn.microsoft.com/en-us/library/ffdk7eyz(v=vs.110).aspx) to the nearest integer. The more you learn! I still think this code won't handle boundary cases very well, but it might not matter because of "no man's land." For example, if a PTM had a mass of 0.5, its bin wouldn't have all of its nearby neighbors. 0.499 would be missing. Maybe that's not a case we need to worry about, though.